.github/workflows/release.yml

```
@@ -0,0 +1,111 @@
+name: Release on Version Change
+
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - 'nixfmt.cabal'
+
+jobs:
+  release-if-version-changed:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout main
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 2  # need previous commit for comparison
+
+      - name: Get version from nixfmt.cabal
+        id: get_version
+        run: |
+          CABAL_FILE="nixfmt.cabal"
+
+          # Extract the version
+          VERSION=$(grep -m1 "^version:" $CABAL_FILE | awk '{print $2}')
+          echo "Version found in cabal: $VERSION"
+
+          # Get previous version from last commit
+          PREV_VERSION=$(git show HEAD~1:"$CABAL_FILE" | grep -m1 "^version:" | awk '{print $2}' || echo "none")
+          echo "Previous version: $PREV_VERSION"
+
+          # Export for later steps
+          echo "version=$VERSION" >> $GITHUB_OUTPUT
+          echo "prev_version=$PREV_VERSION" >> $GITHUB_OUTPUT
+
+      - name: Check if version changed
+        id: compare
+        run: |
+          if [ "${{ steps.get_version.outputs.version }}" != "${{ steps.get_version.outputs.prev_version }}" ]; then
```

It's important to pass inputs via env vars instead of directly interpolating into the shell script. Here's how:

```suggestion
        env:
          PREV_VERSION: ${{ steps.get_version.outputs.prev_version }}
          VERSION: ${{ steps.get_version.outputs.version }}
        run: |
          if [ "$PREV_VERSION" != "$VERSION" ]; then
```

Similar other interpolations below.